### PR TITLE
Add workload for odo summit lab

### PR DESCRIPTION
**Pull request by jankleinert**
##### SUMMARY
Adds a workload for the odo summit lab

##### ISSUE TYPE
- New role Pull Request

### CODE REVIEW

File: ansible/roles/ocp-workload-odo-workshop/tasks/pre_workload.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+---
+- name: Copy .kube/config and set env var
+  copy:
+    src: ~/.kube/config
+    dest: /tmp/${GUID}
+  environment: 
+    KUBECONFIG: /tmp/${GUID}
```
Comment (fridim) [inline ansible/roles/ocp-workload-odo-workshop/tasks/pre_workload.yml]: `${GUID}` is syntax error here. It's jinja template, not Shell.

Plus, environment is useless here.

I would simply add a variable in `defaults/main.yml`:

```yaml
tmp_kubeconfig: /tmp/{{ guid }}
```
Your `copy` task would become:

```yaml
- name: Copy .kube/config and set env var
  copy:
    src: ~/.kube/config
    dest: "{{ tmp_kubeconfig }}"
```
Then you have to add this part:

```
  environment:
    KUBECONFIG: "{{ tmp_kubeconfig }}"
```
to every tasks that run oc commands.

You can factorize using `block` like this:

```yaml
- environment:
    KUBECONFIG: "{{ tmp_kubeconfig }}"
  block:
     - task1
     - task2
     ...
```

File: ansible/roles/ocp-workload-odo-workshop/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,30 @@
+---
+- name: Running Pre Workload Tasks
+  import_tasks: ./pre_workload.yml
```
Comment (fridim) [inline ansible/roles/ocp-workload-odo-workshop/tasks/main.yml]: `./pre_workload.yml` -> can be simply `pre_workload.yml`

File: ansible/roles/ocp-workload-odo-workshop/tasks/remove_workload.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
+---
+- name: post_workload Tasks Complete
+  debug:
+    msg: "Pre-Software checks completed successfully - Removed"
+
+- name: Remove odo workshop resources
+  shell: "oc delete \
+  services,routes,deploymentconfigs,imagestreams,secrets,configmaps,serviceaccounts,rolebindings,serviceaccounts,rolebindings,persistentvolumeclaims,pods \
+  --selector 'app={{odo_application_name}}-{{odo_project_name}},spawner=learning-portal'"
```
Comment (fridim) [inline ansible/roles/ocp-workload-odo-workshop/tasks/remove_workload.yml]: use YAML multiline:

```yaml
- name: Remove odo workshop resources
  shell: >-
    oc delete
    services,routes,deploymentconfigs,imagestreams,secrets,configmaps,serviceaccounts,rolebindings,serviceaccounts,rolebindings,persistentvolumeclaims,pods
    --selector 'app={{odo_application_name}}-{{odo_project_name}},spawner=learning-portal'
```

File: ansible/roles/ocp-workload-odo-workshop/tasks/remove_workload.yml [hunk 4]
```diff
@@ -0,0 +1,24 @@
...
+  ignore_errors: true
+
+- name: Remove odo workshop cluster resources
+  shell: "oc delete \
+  clusterrolebindings,clusterroles \
+  --selector 'app={{odo_application_name}}-{{odo_project_name}},spawner=learning-portal'"
```
Comment (fridim) [inline ansible/roles/ocp-workload-odo-workshop/tasks/remove_workload.yml]: same, multiline yaml `>-` is handy

File: ansible/roles/ocp-workload-odo-workshop/tasks/workload.yml [hunk 5]
```diff
@@ -0,0 +1,21 @@
...
+- name: Create project for odo workshop if it doesn't exist
+  shell: "oc new-project {{odo_project_name}} --display-name='odo workshop'"
+  ignore_errors: true
+
+- name: "Create deployment"
+  command: "oc new-app https://raw.githubusercontent.com/openshift-labs/workshop-spawner/3.0.3/templates/learning-portal-production.json \
+   --param PROJECT_NAME={{odo_project_name}} \
+   --param APPLICATION_NAME={{odo_application_name}} \
+   --param TERMINAL_IMAGE={{odo_terminal_image}} \
+   --namespace {{odo_project_name}}"
```
Comment (fridim) [inline ansible/roles/ocp-workload-odo-workshop/tasks/workload.yml]: multiline yaml `>-` candidate

File: ansible/roles/ocp-workload-odo-workshop/tasks/remove_workload.yml [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+  ignore_errors: true
+
+- name: Remove odo workshop project
+  shell: "oc delete project {{odo_project_name}}"
+  ignore_errors: true
```
Comment (fridim) [inline ansible/roles/ocp-workload-odo-workshop/tasks/remove_workload.yml]: i would not put `ignore_errors: true` everywhere

You can check for existence of the project, if it exists, then delete.

File: ansible/roles/ocp-workload-odo-workshop/tasks/remove_workload.yml (hunk 3 above)
Comment (jankleinert) [reply ansible/roles/ocp-workload-odo-workshop/tasks/remove_workload.yml]: Thanks! I didn't even know that was possible.

**Comment by jankleinert** [comment]
@fridim Thanks for looking at this - I'm pretty new to ansible, as you can probably tell! I think I've addressed all your comments. I tested this against a non-shared RHPDS  cluster using become_override=true and against one of the shared clusters without setting become_override to true and it seems to work as expected in both cases.

**Comment by fridim** [comment]
merged, thanks!

**Pull request merged**